misc/python/materialize/checks/scenarios_upgrade.py

```diff
@@ -509,3 +510,69 @@ def actions(self) -> list[Action]:
             ),
             Validate(self),
         ]
+
+
+class SelfManagedv25_2_Upgrade(Scenario):
+    """
+    Upgrade from the oldest v25.2 patch release to the latest v25.2 patch release.
+    """
+
+    def __init__(
+        self,
+        checks: list[type[Check]],
+        executor: Executor,
+        features: Features,
+        seed: str | None = None,
+    ):
+        self_managed_versions = fetch_self_managed_versions()
+        self.v25_2_versions = sorted(
+            [
+                v.version
+                for v in self_managed_versions
+                if v.helm_version.major == 25 and v.helm_version.minor == 2
+            ]
+        )
+
+        super().__init__(checks, executor, features, seed)
+
+    def base_version(self) -> MzVersion:
+        return self.v25_2_versions[0]
+
+    def actions(self) -> list[Action]:
+        print(f"Upgrading from tag {self.base_version()}")
+
+        def upgrade_actions(version: MzVersion | None, generation: int) -> list[Action]:
+            previous_service_name = f"mz_{generation - 1}"
+            service_name = f"mz_{generation}"
+            return [
+                start_mz_read_only(
+                    self,
+                    tag=version,
+                    deploy_generation=generation,
+                    mz_service=service_name,
+                ),
+                WaitReadyMz(service_name),
+                PromoteMz(service_name),
+                Validate(self, mz_service=service_name),
+                KillMz(capture_logs=True, mz_service=previous_service_name),
+            ]
+
+        actions = [
+            StartMz(
+                self,
+                tag=self.base_version(),
+                mz_service="mz_1",
+            ),
+            Initialize(self, mz_service="mz_1"),
+            Manipulate(self, phase=1, mz_service="mz_1"),
```

Comment:
It would be great if we could run upgrades inbetween the Initialize/Manipulate1/Manipulate2 phases too.